packages/vouching/contracts/Vouching.sol

```diff
@@ -0,0 +1,99 @@
+pragma solidity ^0.4.24;
+
+
+import "openzeppelin-zos/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-zos/contracts/token/ERC20/SafeERC20.sol";
+import "openzeppelin-zos/contracts/math/SafeMath.sol";
+import "zos-lib/contracts/Initializable.sol";
+
+contract Vouching is Initializable {
+  event OwnershipTransferred(address indexed oldOwner, address indexed newOwner);
+  event DependencyCreated(string name, address indexed owner, address indexed dependencyAddress, uint256 initialStake);
+  event Vouched(string name, uint256 amount);
+  event Unvouched(string name, uint256 amount);
+  event DependencyRemoved(string name);
+
+  using SafeMath for uint256;
+  using SafeERC20 for ERC20;
+
+  struct Dependency {
+    address owner;
+    address dependencyAddress;
+    uint256 stake;
+  }
+
+  mapping (string => Dependency) private _registry;
+  uint256 private _minimumStake;
+  ERC20 private _token;
+
+  modifier onlyDependencyOwner(string name) {
+    require(msg.sender == _registry[name].owner, "Sender must be the dependency owner");
+    _;
+  }
+
+  function initialize(uint256 minimumStake, ERC20 token) initializer public {
+    require(token != address(0), "Token address cannot be zero");
+    _minimumStake = minimumStake;
+    _token = token;
+  }
+
+  function minimumStake() public view returns(uint256) {
+    return _minimumStake;
+  }
+
+  function token() public view returns(ERC20) {
+    return _token;
+  }
+
+  function getDependency(string name) public view returns(address, address, uint256) {
+    return (
+      _registry[name].dependencyAddress,
+      _registry[name].owner,
+      _registry[name].stake
+    );
+  }
+
+  function create(string name, address owner, address dependencyAddress, uint256 initialStake) external {
+    require(initialStake >= _minimumStake, "Initial stake must be equal or greater than minimum stake");
+    require(owner != address(0), "Owner address cannot be zero");
+    require(dependencyAddress != address(0), "Dependency address cannot be zero");
+    require(_registry[name].dependencyAddress == address(0), "Given dependency name was already registererd");
+
+    _token.safeTransferFrom(owner, this, initialStake);
+
+    _registry[name] = Dependency(owner, dependencyAddress, initialStake);
+
+    emit DependencyCreated(name, owner, dependencyAddress, initialStake);
```

Comment:
One last thing here, following the [check-effects-interactions pattern](https://solidity.readthedocs.io/en/v0.4.25/security-considerations.html?highlight=effects#use-the-checks-effects-interactions-pattern), the `transferFrom` should be the last part of this function

Reply:
Okey, good to know. Thanks facu. Should be fixed now.